lib/core/src/display/shape/text/text_field/keyboard.rs

```diff
@@ -0,0 +1,197 @@
+//! A FRP definitions for keyboard event handling, with biding this FRP graph to js events.
+
+use crate::prelude::*;
+
+use crate::display::shape::text::text_field::cursor::Step;
+use crate::display::shape::text::text_field::TextFieldData;
+use crate::system::web::text_input::KeyboardBinding;
+
+use enso_frp::*;
+use web_sys::KeyboardEvent;
+
+
+
+// ====================
+// === TextFieldFrp ===
+// ====================
+
+/// This structure contains all nodes in FRP graph handling keyboards events of one TextField
+/// component.
+///
+/// The most of TextField actions are covered by providing actions to KeyboardActions for specific
+/// key masks. However, there are special actions which must be done in a lower level:
+///  * *clipboard operations* - they are performed by reading text input js events directly from
+///    text area component. See `system::web::text_input` crate.
+///  * *text input operations* - here we want to handle all the keyboard mapping set by user, so
+///    we connect this action directly to `key_press` node from `keyboard`.
+#[derive(Debug)]
+pub struct TextFieldFrp {
+    /// A "keyboard" part of graph derived from frp crate.
+    keyboard: Keyboard,
+    /// Keyboard actions. Here we define shortcuts for all actions except letters input, copying
+    /// and pasting.
+    actions: KeyboardActions,
+    /// A node producing event once cut operation was requested.
+    cut: Dynamic<()>,
+    /// A node producing event once copy operation was requested.
+    copy: Dynamic<()>,
+    /// A node producing event once paste operation was requested.
+    paste: Dynamic<String>,
+    /// A lambda node performing cut operation. Returns the string which should be copied to
+    /// clipboard.
+    cut_action: Dynamic<String>,
```

Comment:
This name is not clear. Should it be named `on_cut`? This comment applies to everything with suffix `_action`

Reply:
For me, the on_cut is misleading. It suggest that the point of this node is to emit events. And that's not the case, because this node is rather consuming event doing some action.